src/components/Badge/Badge.tsx

```diff
@@ -48,23 +54,44 @@ const Badge = (props: BadgeProps) => {
     if (fill) {
       classes += "-fill";
     }
+
+    switch (size) {
+      case "small":
+        classes += " badge-small";
+        break;
+      case "medium":
+        classes += " badge-medium";
+        break;
+      case "large":
+        classes += " badge-large";
+        break;
+      default:
+        classes += " badge-medium";
+        break;
+    }
     setClasses(classes);
-  }, [state, fill]);
+  }, [state, fill, size]);
 
   return (
-    <div
-      style={style ?? {}}
-      className={`nitrozen-badge ${classes}`}
-      onClick={onClickHandler}
-      {...restProps}
-    >
-      {labelText}
+    <div className={kind === "service" ? "badge-background" : ""}>
```

Comment:
Move this expression as 
`const containerClassname = useMemo(() => (kind === "service" ? "badge-background" : ""), [kind])`